src/native/external/zlib/adler32.c

```diff
@@ -5,6 +5,11 @@
 
 /* @(#) $Id$ */
 
+#ifdef _MSC_VER
+__pragma(warning (disable:4005)) /* macro redefinition */
+__pragma(warning (disable:4131)) /* uses old-style declarator */
+#endif
```

Comment:
we should probably try to upstream these to https://github.com/madler/zlib so we can link the PR in https://github.com/dotnet/runtime/blob/main/src/native/external/zlib-version.txt like we do for https://github.com/dotnet/runtime/blob/main/src/native/external/libunwind-version.txt

Reply:
Something we could do after we merged this PR? Don't want to delay this PR more if not really necessary.

Reply:
Moved disabling hof zlib warnings from source files into cmake files.